app/models/effective/tagging.rb

```diff
@@ -0,0 +1,41 @@
+module Effective
+  class Tagging < ApplicationRecord
+    self.table_name = (EffectivePages.taggings_table_name || :taggings).to_s
+
+    belongs_to :tag, class_name: 'Effective::Tag'
+    belongs_to :taggable, polymorphic: true
+
+    validates :tag,      presence: true
+    validates :taggable, presence: true
+
+    scope :deep, -> { all }
+
+    effective_resource do
+      tag_id          :integer
+      taggable_id     :integer
+      taggable_type   :string
```

Comment:
If we're using `tag.name` for the to_s.  Could we denormalize?  Add a `tag_name :string` column and a `before_validation self.tag_name = tag.to_s`